### DISCUSSION TITLE
Big performance difference between iphone 5se and samsung galaxy s8

### DISCUSSION THREAD

**Discussion by xzilja** [pixijs/pixijs#7255] [General]
I was testing my app on iPhone 5s where it holds steady 60fps without any issues and Samsung galaxy s8 where it struggled to hit 25fps in some places.

After some tinkering I was able to achieve decent performance on samsung by applying "A LOT" cacheAsBitmap's. So I assume performance is hindered by amount of objects in a scene? But to be honest I don't think I have that many (see images below). I am using BitmapFonts, avoiding filters, and each scene has max of like 2 square masks. My sprites are packed into spritsheets as well I have roughly 5 4000x4000 spritesheets.

So I was wondering why such big performance diff between ios flagship released in 2016 and android one released in 2017?

My settings for pixi v6 look like this
```ts
PIXI.settings.RESOLUTION = devicePicelRatio;

const danteApp = new PIXI.Application({
  width: innerWidth,
  height: innerHeight,
  autoDensity: true,
  view: document.getElementById('my-canvas') as HTMLCanvasElement,
  powerPreference: 'high-performance',
  antialias: false,
  backgroundAlpha: 0
});
```

Most troublesome screens are ones with scrollable lists (note each element that is not visible on the screen is culled)
<img width="358" alt="Screenshot 2021-02-20 at 16 04 28" src="https://user-images.githubusercontent.com/3154053/108598297-7bd7e580-7395-11eb-840d-da6c28deb814.png">

**Comment by bigtimebuddy** [comment]
I think you need a more nuanced approach for resolution other than `devicePixelRatio`. Those two devices have very different ppi's. iPhone 5 is 325ppi and Galaxy S8 is 570ppi. This means you're creating A LOT more pixels on S8. You may want to consider capping the `RESOLUTION` to something like `2`. You're trading off a little sharpness for stability. Seems worth it.

**Reply by xzilja** [reply]
I considered that, I even tested with resolution of 1, lag still persisted on samsung device. So far best I can tell is that it can't handle many instances of various pixie elements visible on screen at once, even with small resolution. The only solution I founc to work was that cacheAsBitmap flag.

Not an issue with pixi, but I am just curious to why this is happening. Maybe it is something specific to GPU / WebGL there? Not sure.

**Reply by bigtimebuddy** [reply]
CacheAsBitmap doesn’t give us a ton of information. Does reducing the spritesheet sizes by 25% make any difference? Does turning visibility off on some elements help?
